pkg/client/apiutil/lazyrestmapper.go

```diff
@@ -199,26 +211,32 @@ func (m *lazyRESTMapper) addKnownGroupAndReload(groupName string, versions ...st
 
 // findAPIGroupByName returns API group by its name.
 func (m *lazyRESTMapper) findAPIGroupByName(groupName string) (metav1.APIGroup, error) {
-	// Ensure that required info about existing API groups is received and stored in the mapper.
-	// It will make 2 API calls to /api and /apis, but only once.
-	if m.apiGroups == nil {
-		apiGroups, err := m.client.ServerGroups()
-		if err != nil {
-			return metav1.APIGroup{}, fmt.Errorf("failed to get server groups: %w", err)
-		}
-		if len(apiGroups.Groups) == 0 {
-			return metav1.APIGroup{}, fmt.Errorf("received an empty API groups list")
+	// Looking in the cache first.
+	for i := range m.apiGroups {
+		if groupName == (&m.apiGroups[i]).Name {
```

Comment:
Could we assign `m.apiGroups[i]` to a local variable?

Reply:
Yeah, totally makes sense. Thanks for the suggestion!